### ISSUE TITLE
Feature Request: Add support for HTTP2 to azurerm_api_management

### ISSUE THREAD

**Issue by NickMetz** [terraform-providers/terraform-provider-azurerm#3621]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
It should be possible to enable HTTP/2 Client-side protocol in azurerm_api_management (default it is disabled). 

![apmi](https://user-images.githubusercontent.com/26707368/59165102-a973f100-8b16-11e9-94f5-7bca8ecd6d3d.png)

Please be aware that there is currently a bug #3125 in security block, that may will also impact this feature request. 

**UPDATE:**  Maby we can separate this setting to a new block, it seems not to rely on Security see api call.

### Affected Resource(s)

* azurerm_api_management

### Potential Terraform Configuration

```hcl
resource "azurerm_api_management" "test" {
  name                = "example-apim"
  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"
  publisher_name      = "My Company"
  publisher_email     = "company@terraform.io"

  sku {
    name     = "Developer"
    capacity = 1
   }
  
   protocols {
      enable_http2 = false
   }
}

```
This is how it looks like on API (default is false)
```
    ],
    "privateIPAddresses": null,
    "additionalLocations": null,
    "virtualNetworkConfiguration": null,
    "customProperties": {
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Ssl30": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls10": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls11": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TripleDes168": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Ssl30": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls10": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls11": "false",
      "Microsoft.WindowsAzure.ApiManagement.Gateway.Protocols.Server.Http2": "False"
    },
    "virtualNetworkType": "None",
    "certificates": []
```

### References

* https://azure.microsoft.com/en-us/updates/http-2-support-in-azure-api-management/
* https://www.terraform.io/docs/providers/azurerm/r/api_management.html#security

**Comment by alanwales** [comment]
Any update on this? Whenever we update an api with terraform it spends half an hour reverting the http2 enabled setting to false then we have to change it back with PowerShell afterwards

**Comment by NickMetz** [comment]
Released in 1.44.0
